symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/AbstractDatafeedService.java

```diff
@@ -59,155 +53,154 @@ public AbstractDatafeedService(ApiClient agentClient, ApiClient podClient, AuthS
      * {@inheritDoc}
      */
     @Override
-    public void subscribe(DatafeedEventListener listener) {
+    public void subscribe(RealTimeEventListener listener) {
         listeners.add(listener);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public void unsubscribe(DatafeedEventListener listener) {
+    public void unsubscribe(RealTimeEventListener listener) {
         listeners.remove(listener);
     }
 
     /**
-     * Handle a received listener by using the subscribed {@link DatafeedEventListener}.
+     * Handle a received listener by using the subscribed {@link RealTimeEventListener}.
      *
      * @param events List of Datafeed events to be handled
      *
-     * @throws ApiException if the bot cannot get the information about itself by using {@link BotInfoService}
      */
-    protected void handleV4EventList(List<V4Event> events) throws ApiException {
+    protected void handleV4EventList(List<V4Event> events) {
         for (V4Event event : events) {
             if (event == null || event.getType() == null) {
                 continue;
             }
             if (this.isSelfGeneratedEvent(event)) {
                 continue;
             }
-            for (DatafeedEventListener listener : listeners) {
+            for (RealTimeEventListener listener : listeners) {
                 switch (event.getType()) {
                     case DatafeedEventConstant.MESSAGESENT:
                         listener.onMessageSent(event);
                         if (event.getPayload() != null) {
-                            listener.onMessageSent(event.getPayload().getMessageSent());
+                            listener.onMessageSent(event.getInitiator(), event.getPayload().getMessageSent());
                         }
                         break;
```

Comment:
This `if` condition is not necessary. Our API contract ensure that this field will always be part of the response payload so I'd modify this `case` (as well as the following ones) to : 
```java
case: 
    listener.onMessageSent(event);
    listener.onMessageSent(event.getInitiator(), event.getPayload().getMessageSent());
    break;
```

Reply:
Generated class marked method `getPayload` with annotation `@Nullable`, so it created a warning about NullPointerException and in Intellij, it looks quite annoying. 😞

Reply:
Moving to an _enum-based_ pattern (cf. [here](https://github.com/SymphonyOSF/symphony-sdk-java-v2/blob/master/symphony-sdk-api/src/main/java/com/symphony/sdk/api/datafeed/DatafeedEventType.java#L15)) would fix this issue. 

I start having some doubts regarding the fact that we should return the full `V4Event`. This object contains all the different payloads (`onMessageSent`, `onMessageSuppressed`, etc.) and this might create some confusion for some developers.